pydantic_source/source.py

```diff
@@ -0,0 +1,154 @@
+import ast
+from typing import Any, Type, Dict, Iterable
+from benedict import benedict
+from pathlib import Path
+
+from pydantic_settings import BaseSettings, PydanticBaseSettingsSource
+from pydantic.fields import FieldInfo
+
+from rapyuta_io_sdk_v2 import Client, Configuration
+import base64
+
+
+class ConfigTreeSource(PydanticBaseSettingsSource):
+    def __init__(
+        self,
+        settings_cls: Type["BaseSettings"],
+        config: Configuration,
+        tree_name: str = "default",
+        key_prefix: str = "",
+        api_with_project: bool = True,
```

Comment:
nitpick: Let's keep it same as the SDK.
```suggestion
        with_project: bool = True,
```